src/renderer/services/file-management-system/Md5Hasher.ts

```diff
@@ -1,46 +1,70 @@
-import { ArrayBuffer } from "spark-md5";
+import { createMD5 } from "hash-wasm";
+import { IHasher } from "hash-wasm/dist/lib/WASMInterface";
+
 
 export default class Md5Hasher{
 
-    private hasher: ArrayBuffer
+    private static readonly ARRAY_DELIMITER = ",";
+
+    private hasher: IHasher
+
+    public constructor(hasher: IHasher) {
+        this.hasher = hasher;
+    }
+
 
-    public constructor (hasher?: ArrayBuffer) {
-        this.hasher  = hasher || new ArrayBuffer();
+    /**
+     * Factory method for creating Md5Hashers
+     * @returns Promise<Md5Hasher>
+     */
+     public static async init(): Promise<Md5Hasher>{ 
+        const hasher = await createMD5();
+        hasher.init();
+        return new Md5Hasher(hasher);
     }
 
 
     /**
-     * Converts serialized string to SparkMD5.State, having the same state as when it was serialized.
+     * Converts serialized string to Uint8Array, having the same state as when it was serialized.
      * @param serialized_md5 
-     * @returns 
+     * @returns Promise<Md5Hasher>
      */
-     public static deserialize (serialized_md5: string) : Md5Hasher{ 
-        const hasher = new ArrayBuffer();
-        hasher.setState(JSON.parse(serialized_md5));
+     public static async deserialize(serializedMd5: string): Promise<Md5Hasher>{
+        // The load() function necessary to rehydrate the Md5Hasher state requires a Uint8Array,
+        // but in order to persist the Uint8Array it had to be converted to string.
+        // This intends to take an array-like string and convert it back into an array
+        // that a Uint8Array can be created from.
+        const serializedMd5AsArray = serializedMd5.split(Md5Hasher.ARRAY_DELIMITER).map(v => +v);
```

Comment:
This looks like it maps each delimited string v using `(v => +v)`.  Im a bit confused by that, can you explain?

Reply:
The `+` converts the number-like string into an integer